sound/soc/sof/loader.c

```diff
@@ -372,6 +372,8 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 				 msecs_to_jiffies(sdev->boot_timeout));
 	if (ret == 0) {
 		dev_err(sdev->dev, "error: firmware boot failure\n");
+		/* after this point FW_READY msg should be ignored */
+		sdev->boot_complete = true;
```

Comment:
Humm, not quite getting the idea. 
the boot_complete is tested for in snd_sof_ipc_msgs_rx() which is called in the interrupt thead, so isn't there a risk here? so you *could* have a case where you timeout, and before setting boot_complete to true you deal with the interrupt.
I guess if @lyakh approves I am fine but I can't figure this one out by myself...

Reply:
@plbossart wrote:
> the boot_complete is tested for in snd_sof_ipc_msgs_rx() which is called in the interrupt thead, > so isn't there a risk here? so you _could_ have a case where you timeout, and before setting 
> boot_complete to true you deal with the interrupt.

This is ok, if interrupt occurs here we are still all set up and FW_READY can be handled. We set up the debugfs nodes and what not but no kernel crashes. 

The problem I hit if the interrupt happens after this point. E.g. even snd_sof_ipc_free() still doesn't actually disable the IPC, so there's quite a wide time window where resources have been freed while the IPC irq is still active and can potentially receive FW_READY.

Reply:
ok. If @lyakh agrees then we are good to go.

Reply:
I will assume @lyakh is fine with the changes...